tools/ci/docker/linux/Dockerfile

```diff
@@ -68,6 +69,31 @@ RUN mkdir -p $CARGO_HOME \
   && $CARGO_HOME/bin/rustup target add thumbv6m-none-eabi \
   && $CARGO_HOME/bin/rustup target add thumbv7m-none-eabi
 
+
+# Install Zig from NuttX
+ENV ZIG_HOME=/tools/zig
+
+# Stable
+#RUN apt-get install -y -qq \
+#  python3 \
+#  python3-pip \
+#  && pip3 install ziglang \
+#  && python3 -m ziglang version
+
+# Nightly
+RUN \
+    mkdir -p $ZIG_HOME/lib/zig && mkdir $ZIG_HOME/bin && \
+    curl -O -s $(curl -s https://ziglang.org/download/index.json \
+    | jq ".master.\"$(uname -m)-$(uname | awk '{ print tolower($0) }')\"" \
+    | jq -r .tarball); \
+    file=$(ls | grep '.tar.xz') && \
+    tar -xf $file &&\
+    folder=${file%.*.*} &&\
+    mv $folder/lib/* $ZIG_HOME/lib/zig/ &&\
+    mv $folder/zig $ZIG_HOME/bin/ &&\
+    rm -rf $folder && \
+    echo "Zig version: $($ZIG_HOME/bin/zig version)\n Zig/Clang version: $($ZIG_HOME/bin/zig cc --version)"
```

Comment:
```suggestion
RUN mkdir -p $ZIG_HOME/lib/zig && mkdir $ZIG_HOME/bin \
  && curl -O -s $(curl -s https://ziglang.org/download/index.json \
  | jq ".master.\"$(uname -m)-$(uname | awk '{ print tolower($0) }')\"" \
  | jq -r .tarball) \
  && file=$(ls | grep '.tar.xz') \
  && tar -xf $file \
  && folder=${file%.*.*} \
  && mv $folder/lib/* $ZIG_HOME/lib/zig/ \
  && mv $folder/zig $ZIG_HOME/bin/ \
  && rm -rf $folder
```

Reply:
BTW why we can't use `curl -sL https://ziglang.org/download/0.9.1/zig-linux-x86_64-0.9.1.tar.xz | tar -xJC $(ZIG_HOME) --strip-components=1` ?

Reply:
In docker there would be a continuous update without having to specify versions manually. Since zig has no updater.

Reply:
But if that's the case, I can change it. It would be necessary?

Reply:
I'm not sure if we have a consumer for that `echo` during the Docker image rebuild

Reply:
think that it is better to have tools version controlled, so it will not pick tools randomly from one image rebuild to another

Reply:
changed